File: polly.rb
require "sinatra"
require "json"
require "polly_phone"

configure do
  set :root, File.dirname(__FILE__)
  set :public_folder, "public/app"
end

def get_config
  PollyPhone::Site.new(File.dirname(__FILE__) + "/sites/config.yml")
end

get "/" do
  File.read("public/app/index.html")
end

get "/brands" do
  JSON brands: get_config.brands_list
end

get "/models" do
  JSON models: get_config.models(params[:brand]) || nil
end

get "/phones" do
  JSON desc: get_config.phone_desc(params[:phone_url])
end

get "/search" do
  JSON items: get_config.search(params[:item])
end

File: public/app/js/polly.js
var app = angular.module('pollyApp', []);

app.controller('phonesController', function($scope, $http) {
      
    $http.get('/brands').success(function(data) {
        $scope.brands = data.brands;
    });

    $scope.getModels = function () {
      $http.get('/models', { params: { brand: $scope.selectedBrand } }).success(function(data) {
         $scope.models = data.models;
      });
    };

    $scope.getDesc = function () {
      $http.get('/phones', { params: { phone_url: $scope.selectedModel.link } }).success(function(data) {
         $scope.description = data.desc;
      });
    };

});

app.controller('searchController', function($scope, $http) {
     
    $scope.searchItem = function () {
      $http.get('/search', { params: { item: $scope.searchedItem } }).success(function(data) {
         $scope.items = data.items;
      });
    };
});